backend/core/src/aggregator.rs

```diff
@@ -124,10 +124,19 @@ pub struct AddNode {
     pub node: NodeDetails,
     /// Genesis [`Hash`] of the chain the node is being added to.
     pub genesis_hash: Hash,
-    /// Connection id used by the node connector for multiplexing parachains
-    pub conn_id: ConnId,
-    /// Address of the NodeConnector actor
-    pub node_connector: Addr<NodeConnector>,
+    /// Source from which this node is being added (Direct | Shard)
+    pub source: NodeSource,
```

Comment:
dq: why not put the source in `NodeDetails`?

Reply:
`NodeDetails` needs to be serializable, while `NodeSource` includes a ref to an actor.